Add Prompting 101 Tile to independent-ai-topics (#1)
Ada-Developers-Academy/independent-ai-topics: 02-prompting-101/making-requests-from-chatgpt.md

| Prompt Engineering | The process of structuring text to describe tasks to be more easily processed and understood by generative AI tools.  | | “My initial prompt didn’t return what I expected, so I’m practicing a few prompt engineering techniques and rewriting my prompt to get more complete results.” |

## How are People Using AI Assistance and Generative Tools?

The readings from the Generative AI Overview lesson outlined a handful of uses for Large Language Models in the business world, but how can they help us, right now, in our own lives? We’ve listed some more use cases below with examples of how folks have used them, but there are a couple caveats to think about before we dive into our own prompting:
- Like any website with unknown sources, if an AI tool like ChatGPT is generating concrete information like dates and names that we did not provide it, we want to do our own web searches to confirm those details are accurate and not fabricated or otherwise false. We’ll dive more into techniques for verifying responses shortly.

> - There are legal considerations if we use AI tools for public work or businesses. Many tools have been trained on massive data sets with text, art and other works scraped from the internet without legal clearance. Unless a tool has been built solely with data that has been given permission to be used for training and generation, it may not be legal to use the art or text generated by an AI tool verbatim. 

Reviewer: The last sentence was a bit difficult for me to parse the first time reading it. What do you think of this small change?

```suggestion
- There are legal considerations if we use AI tools for public work or businesses. Many tools have been trained on massive data sets with text, art and other works scraped from the internet without legal clearance. Using the art or text generated by an AI tool verbatim may be legally restricted unless the tool has been exclusively constructed with data that has been authorized for training and generation.
```

Author: That reads much clearer to me too, thanks!

| Categorizing sentiment from text | If we ran a website or moderated for a forum, we could ask for user feedback to be categorized to learn which areas are painful to users. |
| Formatting information | Creating a table of javascript syntax for a study guide |
| Generating quizzes for study | Given notes on a particular subject, generate 5 questions based on the notes |
| First drafts of letters or emails | Given a list of information to include, we can ask an AI tool to craft the letter, then review and customize it to our needs. |
| Reviewing text or resumes for spelling, grammar, & consistency | AI tools can be an extra set of eyes for language features like punctuation as well as consistency in how you format information across the text. |
| Generating images with specific AI tools like DALL-E, Midjourney, Photoshop’s Firefly, and more | Creating art to go with a story for a personal gift. Generating a wallpaper with handy python commands |

## Accessing ChatGPT

We're readied with some background on what generative AI tools are and how we can use them, so let's sign up for ChatGPT to get hands-on experience!

### Signing up

> 1. Our first step is to visit [chat.openai.com](chat.openai.com) and press the "Sign Up" button to start creating an account. 

Reviewer: ```suggestion
1. Our first step is to visit [chat.openai.com](https://chat.openai.com) and press the "Sign Up" button to start creating an account. 
```

![chat.openai.com landing page](./assets/chatgpt-landing-page.png)  
*Fig. ChatGPT landing page*

2. There are a few options, you can sign up with existing Google, Microsoft, or Apple accounts by using their buttons to follow their authentication flows, or you can enter an email in the "Email address" field and press continue.

![Sign Up options](./assets/create-account-page.png)  
*Fig. Options to sign up for ChatGPT*

3. If you choose to sign up by email, you'll be taken to a screen where you can update your email if necessary and choose a password.

![Create password screen](./assets/set-password-page.png)  
*Fig. Create password screen for email sign up flow*

4. Once you submit your password, you'll be taken to a notification to check your email for a verification message. You'll need to open the link in the email they sent to verify and finish setting up your account.

![Email verification message](./assets/verify-email-message.png)  
*Fig. Email verification message*

> 5. You can log in with your new account at [chat.openai.com](chat.openai.com) once your email is verified. The first time you log in, you'll be asked to enter your first name, last name, and birth date to verify your age.

Reviewer: ```suggestion
5. You can log in with your new account at [chat.openai.com](https://chat.openai.com) once your email is verified. The first time you log in, you'll be asked to enter your first name, last name, and birth date to verify your age.
```

![Submitting info on first log in](./assets/info-after-login.png)  
*Fig. First log-in information form*

6. After pressing Submit, you will be asked to enter a phone number that can be used to verify future log ins. 

![Submit phone number](./assets/submit-phone-number.png)  
*Fig. Form to submit phone number to receive a verification text*

7. You'll be taken to a new page where you need to enter a verification code after entering your phone number. A text will be sent to the phone number you submitted with your verification code.

![Submit verification code](./assets/phone-verification-code.png)   
*Fig. Form to submit verification code*

> 8. At this point you should be redirected a brand new chat page! We should see "ChatGPT 3.5" at the top, letting us know we are working with OpenAI's 3.5 model. 

Reviewer: ```suggestion
8. At this point you should be redirected to a brand new chat page! We should see "ChatGPT 3.5" at the top, letting us know we are working with OpenAI's 3.5 model. 
```

## More on Regenerating a Response

Let's go back to our initial prompt: "`Write a story about a frog.`" We didn't ask for any constraints on how long the story should be or subject matter other than it had to include a frog. Say that we wanted a different story about a frog – we could add to our prompt if we had specific things to include, but what if we just want another frog story?

One option would be to pose the prompt again again, retype and submit "`Write a story about a frog.`", but as we learned above, there's an even easier way – the "Regenerate" button! When we use the "Regenerate" button, just like when we first sent the prompt, we'll see the text being put together a word or phrase at a time until it's finished. The new response will replace the spot in the chat where the original response was, but the original isn't lost to us. 

If we take a look at the bottom of the response after regenerating, in front of our original toolbar we should see additional text and buttons that look like: `< 2 / 2 >`. 

![Regenerate viewing options](./assets/regenerate-viewing-tools.png)  
*Fig. response tools after generating a second response*

> The first number indicates how many versions of the response there are, and the second number is which version of the response we are currently viewing. The arrows are buttons that let us navigate forwards and backwards through the responses so we can see all of the versions.

Reviewer: ```suggestion
The second number indicates how many versions of the response there are, and the first number is which version of the response we are currently viewing. The arrows are buttons that let us navigate forwards and backwards through the responses so we can see all of the versions.

For example, if we generate a response 3 times, when the indicator shows `< 1 / 3 >`, this means we are viewing the first output of the prompt, `< 2 / 3 >` means we are viewing the second output of the prompt, and so on.
```

Author: Thanks for catching the number/position swap here, the added example is really helpful. 🙏🏻
